File: backend/src/invoice.rb
class Invoice
  # Permissão de leitura para essas variáveis
  attr_reader :billing_address, :shipping_address, :order
    
  # Construtor
  def initialize(attributes = {})
    # seta billing address
    @billing_address = attributes.values_at(:billing_address)
    # seta shipping address
    @shipping_address = attributes.values_at(:shipping_address)
    # seta order
    @order = attributes.values_at(:order)
  end
end

File: backend/invoice.py
class Invoice:

    def __init__(self, order):
        '''
        Constructor for the Invoice class.

        Args:
            order (Order): order to which the invoice is being issued.
        '''
        self.order = order
        self.billing_address = self.order.customer.billing_address
        self.shipping_address = self.order.customer.shipping_address